flutter/lib/src/sentry_asset_bundle.dart

```diff
@@ -0,0 +1,198 @@
+    final span = _hub.getSpan()?.startChild(
+          'file.read',
+          description: 'AssetBundle.load: ${_fileName(key)}',
+        );
+
+    span?.setData('file.path', key);
+
+    ByteData? data;
+    try {
+      data = await _bundle.load(key);
+      _setDataLength(data, span);
+      span?.status = SpanStatus.ok();
+    } catch (exception) {
+      span?.throwable = exception;
+      span?.status = SpanStatus.internalError();
+      rethrow;
+    } finally {
+      await span?.finish();
+    }
+    return data;
+  }
+
+  @override
+  Future<T> loadStructuredData<T>(String key, _Parser<T> parser) {
+    if (_enableStructuredDataTracing) {
+      return _loadStructuredDataWithTracing(key, parser);
+    }
+    return _bundle.loadStructuredData(key, parser);
+  }
+
+  Future<T> _loadStructuredDataWithTracing<T>(
+      String key, _Parser<T> parser) async {
+    final span = _hub.getSpan()?.startChild(
+          'file.read',
+          description: 'AssetBundle.loadStructuredData<$T>: ${_fileName(key)}',
+        );
+    span?.setData('file.path', key);
+
+    final completer = Completer<T>();
+
+    // This future is intentionally not awaited. Otherwise we deadlock with
+    // the completer.
+    // ignore: unawaited_futures
+    runZonedGuarded(() async {
+      final data = await _bundle.loadStructuredData(
+        key,
+        (value) async => await _wrapParsing(parser, value, key, span),
+      );
+      span?.status = SpanStatus.ok();
+      completer.complete(data);
+    }, (exception, stackTrace) {
+      completer.completeError(exception, stackTrace);
+    });
+
+    T data;
+    try {
+      data = await completer.future;
+      _setDataLength(data, span);
+      span?.status = const SpanStatus.ok();
+    } catch (e) {
+      span?.throwable = e;
+      span?.status = const SpanStatus.internalError();
+      rethrow;
+    } finally {
+      await span?.finish();
+    }
+    return data;
+  }
+
+  @override
+  Future<String> loadString(String key, {bool cache = true}) async {
+    final span = _hub.getSpan()?.startChild(
+          'file.read',
+          description: 'AssetBundle.loadString: ${_fileName(key)}',
+        );
+
+    span?.setData('file.path', key);
+    span?.setData('from-cache', cache);
+
+    String? data;
+    try {
+      data = await _bundle.loadString(key, cache: cache);
+      span?.status = SpanStatus.ok();
+    } catch (exception) {
+      span?.throwable = exception;
+      span?.status = SpanStatus.internalError();
+      rethrow;
+    } finally {
+      await span?.finish();
+    }
+    return data;
+  }
+
+  void _setDataLength(dynamic data, ISentrySpan? span) {
+    int? byteLength;
+    if (data is List<int>) {
+      byteLength = data.length;
+    } else if (data is ByteData) {
+      byteLength = data.lengthInBytes;
+    }
+    if (byteLength != null) {
+      span?.setData('file.size', byteLength);
+    }
+  }
+
+  String _fileName(String key) => Uri.parse(key).pathSegments.last;
```

Comment:
That requires if `pathSegments.isNotEmpty` otherwise it throws, similar to `SentryStackTraceFactory` what does, if the `pathSegments` is empty, we can use the raw `key`

Reply:
So this
```suggestion
  String _fileName(String key) => Uri.tryParse(key)?.pathSegments.last ?? key;
```
would be better.

Technically all keys must be a valid uri, otherwise there's no way to load anything, but yeah, there's nothing which would stop you from passing an empty string.

Reply:
indeed, but `pathSegments` still needs to be checked if `isNotEmpty` otherwise `. last` throws, I guess